scripts/submit_finetune_job.py

```diff
@@ -0,0 +1,202 @@
+            if value is not True:
+                new_cmd_parts.append(str(value))
+
+        return ' '.join(new_cmd_parts)
+
+    new_arguments = override_and_reconstruct_command(d1['tasks'][0]['arguments'][0], train_config, unknown_args)
+    
+    model_name = get_model_name(new_arguments)
+    # if model name has /, replace with _
+    model_name = model_name.replace("/", "_")
+    # try given config only has one
+    dataset_name, dataset_mixer, train_file = check_dataset_selection(new_arguments)
+    print(f"Dataset selection is valid.")
+    print(f"Dataset name: {dataset_name}")
+    print(f"Dataset mixer: {dataset_mixer}")
+    print(f"Train file: {train_file}")
+
+    d = copy.deepcopy(d1)
+    d['tasks'][0]['arguments'][0] = new_arguments
+
+    # name and description
+    exp_name = f"open_instruct_finetune_{model_name}_{now}"
+    d['description'] = exp_name
+    d['tasks'][0]['name'] = exp_name
+
+    # WANDB settings
+    for env in d['tasks'][0]['envVars']:
+        if env['name'] == "WANDB_DISABLED":
+            env['value'] = False
+        if env['name'] == "WANDB_PROJECT":
+            env['value'] = wandb_project
+    d['tasks'][0]['envVars'].append({
+        'name': 'WANDB_API_KEY', 'value': wandb_api_key
+    })
+    d['tasks'][0]['envVars'].append({
+        'name': 'WANDB_NAME', 'value': exp_name
+    })
+    d['tasks'][0]['envVars'].append({
+        'name': 'WANDB_RUN_GROUP', 'value': experiment_group
+    })
+
+    # optionally, print to debug config
+    # print(d)
+
+    fn = "configs/beaker_configs/auto_created/{}.yaml".format(exp_name)
+    file = open(fn, "w")
+    yaml.dump(d, file, default_flow_style=True)
+    file.close()
+
+    cmd = f"beaker experiment create {fn} --workspace {args.workspace}"
+    subprocess.Popen(cmd, shell=True)
+
+def check_dataset_selection(command_string):
+    parts = shlex.split(command_string)
+    dataset_name = None
+    dataset_mixer = None
+    train_file = None
+
+    for i, part in enumerate(parts):
+        if part == '--dataset_name' and i + 1 < len(parts):
+            dataset_name = parts[i + 1]
+        elif part == '--dataset_mixer' and i + 1 < len(parts):
+            dataset_mixer = parts[i + 1]
+        elif part == '--train_file' and i + 1 < len(parts):
+            train_file = parts[i + 1]
+
+    if ((dataset_name is not None and dataset_mixer is not None) or
+        (dataset_name is not None and train_file is not None) or
+        (dataset_mixer is not None and train_file is not None)):
+        raise ValueError("Cannot provide two dataset selection mechanisms.")
+
+    return dataset_name, dataset_mixer, train_file
+
+def get_model_name(command_string):
+    parts = shlex.split(command_string)
+    for i, part in enumerate(parts):
+        if part == '--model_name_or_path':
+            if i + 1 < len(parts):
+                return parts[i + 1]
+    return None  # Return None if model name is not found
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
gonna more or less assume this parsing works. would probably be good to test a bit first.